presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/ReorderJoins.java

```diff
@@ -245,8 +227,8 @@ private JoinEnumerationResult chooseJoinOrder(List<PlanNode> sources, List<Symbo
                     .filter(JoinEnumerator::isJoinEqualityCondition)
                     .map(predicate -> toEquiJoinClause((ComparisonExpression) predicate, leftSymbols))
                     .collect(toImmutableList());
-            if (joinConditions.isEmpty() && !isCrossJoinPhase) {
-                return Optional.empty();
+            if (joinConditions.isEmpty()) {
+                return UNKNOWN_COST_RESULT;
```

Comment:
I think we need `INFINITE_COST_RESULT` here instead, see my other comment for https://github.com/Teradata/presto/pull/584/commits/b6e1fa7eaf889f06f9fc1cde73630469662d6123#diff-5cdb22a9680530f837d520839d7eaf99R105